kedro_mlflow/framework/hooks/mlflow_hook.py

```diff
@@ -447,4 +454,15 @@ def on_pipeline_error(
             switch_catalog_logging(catalog, True)
 
 
+def validate_and_sanitize_param_name(name: str) -> str:
+    # regex taken from MLFlow codebase: https://github.com/mlflow/mlflow/blob/e40e782b6fcab473159e6d4fee85bc0fc10f78fd/mlflow/utils/validation.py#L140C1-L148C44
+    pattern = r"^[/\w.\- :]*$"
+
+    if re.match(pattern, name):
+        return name
+    else:
+        # Replace invalid characters with underscore
+        return re.sub(r"[^/\w.\- :]", "_", name)
```

Comment:
can we add a ``logger.warning`` to explain that the parameter is renamed? This might be confusing.